src/ui/x11/x11_ui.h

```diff
@@ -26,7 +30,64 @@ class UiWindowX11 : public UiWindow
     uint32_t _width;
     uint32_t _height;
 
-    std::vector< std::pair<Point2d, uint32_t> > _point;
+    struct PointToDraw
```

Comment:
Hi @fullset could we move these identical structures in a common location as src/ui/ui.h file to avoid code duplication?

Reply:
As I see, there are 2 structures `PointToDraw` and `LineToDraw` which are almost identical in `x11` and `win_ui`. And there are no problem to combine them.

But `EllipseToDraw` structures in `x11` and `win_ui` have different semantics. In `win_ui` we specify top, left, right and bottom coordinates but it `x11` we specify top, left coordinates and width and height. I could make common constructor of 4 doubles but I'm not sure that it's right decision. What do you think?

Reply:
Let's stick to one design which should have top-left coordinate and width with height. For Windows during drawing we could easily find right and bottom values by adding width and height.

Reply:
Hi, @ihhub.

It looks like I've broken windows building by my last changes but I can't find the reason because I haven't windows PC.

Reply:
Let me check...

Reply:
Could you put structures back in ui/ui.h but don't declare these variables:
```cpp
    std::vector<PointToDraw> _point;
    std::vector<LineToDraw> _lines;
    std::vector<EllipseToDraw> _ellipses;
```
?